lib/dap/filter/simple.rb

```diff
@@ -171,6 +171,20 @@ def process(doc)
   end
 end
 
+class FilterFlatten
+  include Base
+  def process(doc)
+    self.opts.each_pair do |k,v|
+      if doc.has_key?(k) and doc[k].is_a?(Hash)
```

Comment:
How about this proposal? Ship this as is, then introduce a flatten_depth later that takes an argument which is the depth length (or, no argument, which flattens the entire thing). I also don't see this operating on arrays (unsure if you were suggesting) -- only hashes since we already have the split_ filters for arrays...